src/chart/util/setDefaults.js

```diff
@@ -0,0 +1,117 @@
+    })
+
+    ////////////////////////////////////////////////////////////
+    //Render single column if no group variable is specified. //
+    ////////////////////////////////////////////////////////////
+    if (!(chart.config.variables.group) || ['', ' '].indexOf(chart.config.variables.group) > -1) {
+        chart.config.variables.group = 'data_all';
+        chart.config.defaults.totalCol = false;
+        chart.config.groups = [{'key':'All'}];
+    }
+
+    ////////////////////////////////////////////////////
+    // Include all group levels if none are specified //
+    ////////////////////////////////////////////////////
+    var groups = d3.set(chart.raw_data.map(d => d[chart.config.variables.group])).values();
+    var groupsObject = groups.map(d => {return {'key': d}; });
+
+    if (!chart.config.groups || chart.config.groups.length === 0)
+        chart.config.groups = groupsObject
+            .sort((a,b) => a.key < b.key ? -1 : a.key > b.key ? 1 : 0);
+
+    //////////////////////////////////////////////////////////////
+    //Check that variables specified in settings exist in data. //
+    //////////////////////////////////////////////////////////////
+    for (var x in chart.config.variables) {
+        var varList = d3.keys(chart.raw_data[0]).concat('data_all');
+
+        if (varList.indexOf(chart.config.variables[x]) === -1) {
+            if (chart.config.variables[x] instanceof Array) {
+                chart.config.variables[x].forEach(function(e) {
+                    if (d3.keys(chart.raw_data[0]).indexOf(e) === -1) {
+                        errorNote('Error in variables object.');
+                        throw new Error(x + ' variable ' + '(\'' + e + '\') not found in dataset.');
+                    }
+                });
+            } else {
+                errorNote('Error in variables object.');
+                throw new Error(x + ' variable ' + '(\'' + chart.config.variables[x] + '\') not found in dataset.');
+            }
+        }
+    }
+    /////////////////////////////////////////////////////////////////////////////////
+    //Check that group values defined in settings are actually present in dataset. //
+    /////////////////////////////////////////////////////////////////////////////////
+    chart.config.groups
+        .forEach(d => {
+            if (groups.indexOf(d.key) === -1) {
+                errorNote('Error in settings object.');
+                throw new Error('\'' + e.key + '\' in the Groups setting is not found in the dataset.');
+            }
+        });
+    
+    /////////////////////////////////////////////////////////////////////////////////
+    //Check that group values defined in settings are actually present in dataset. //
+    /////////////////////////////////////////////////////////////////////////////////
+            if (chart.config.groups.length > 6) {
```

Comment:
No good reason, just seemed about right. Would probably be better to make this a setting actually  `config.defaults.maxGroups` - will make that change.